SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -0,0 +1,171 @@
+//==------- esimd_bit_ops.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu || level_zero
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for esimd default constructor.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <string>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+#include "../logger.hpp"
+#include "../type_coverage.hpp"
+
+using namespace type_coverage;
+using namespace logger;
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental;
+using namespace sycl::ext::intel::experimental::esimd;
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataTypeT, int TestNumber, int NumElems>
+struct kernel_for_simd;
+
+const auto types =
+    named_type_pack<char, unsigned char, signed char, short, unsigned short,
+                    int, unsigned int, long, unsigned long, float, sycl::half,
+                    double, long long, unsigned long long>(
+        {"char", "unsigned char", "signed char", "short", "unsigned short",
+         "int", "unsigned int", "long", "unsigned long", "float", "sycl::half",
+         "double", "long long", "unsigned long long"});
+
+// The main test routine.
+// Use struct with overloaded call operator to have able for iterating over
+// pre-determined datatypes.
+template <typename DataTypeT, int NumElems> struct test {
+  void operator()(const Logger &log, sycl::queue &queue, bool &passed) {
+    DataTypeT default_val{};
+    using AllocatorT = sycl::usm_allocator<DataTypeT, sycl::usm::alloc::shared>;
+
+    // verification for initializing simd by initializer
+    {
+      std::vector<DataTypeT, AllocatorT> result{NumElems, AllocatorT{queue}};
+      queue.submit([&](sycl::handler &cgh) {
+        auto out = result.data();
+        cgh.single_task<kernel_for_simd<DataTypeT, 1, NumElems>>([=] {
```

Comment:
```suggestion
        cgh.single_task<kernel_for_simd<DataTypeT, enum::value, NumElems>>([=] {
```
Magic numbers are going to break some day..

Reply:
I've applied this comment in this commit: 35fd849b6fc222cb2da9d90a1dd6185de07651e7